yarn-project/p2p/src/client/p2p_client.ts

```diff
@@ -126,11 +133,12 @@ export class P2PClient implements P2P {
     private log = createDebugLogger('aztec:p2p'),
   ) {
     const { p2pBlockCheckIntervalMS: checkInterval, p2pL2QueueSize } = getP2PConfigEnvVars();
-    this.blockDownloader = new L2BlockDownloader(l2BlockSource, {
-      maxQueueSize: p2pL2QueueSize,
-      pollIntervalMS: checkInterval,
-    });
-    this.synchedBlockNumber = store.openSingleton('p2p_pool_last_l2_block');
+    const l2DownloaderOpts = { maxQueueSize: p2pL2QueueSize, pollIntervalMS: checkInterval };
+    this.latestBlockDownloader = new L2BlockDownloader(l2BlockSource, l2DownloaderOpts);
+    this.provenBlockDownloader = new L2BlockDownloader(l2BlockSource, { ...l2DownloaderOpts, proven: true });
```

Comment:
I think this is fine for now but I'm thinking tha two downlaoder listening to the same source would mean the same block is read multiple times. 
I'd make the source & downlaoder event emitters/readable streams, but definitely sits in the realm of opimization/nice to haves and can be tackled independently of this PR.

Reply:
Actually it doesn't: the L2Downloader, unlike its name describes, is just a queue that pulls from an Archiver. So each block is only downloaded and processed once.

Reply:
Good point on the remote archiver, hadn't considered it! Agree in that we should revisit this component, and that we should think it so it properly handles reorgs (which is a pending assignment all across the board). But I'd push it for a future PR if that's ok.